scripts/generate-bindata.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+// +build ignore
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"hash/adler32"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/shurcooL/vfsgen"
+)
+
+func needsUpdate(dir string, filename string) bool {
+	oldHash, err := ioutil.ReadFile(filename + ".hash")
+	if err != nil {
+		oldHash = []byte{}
+	}
+
+	adlerHash := adler32.New()
+
+	err = filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		_, _ = adlerHash.Write([]byte(info.Name()))
+		_, _ = adlerHash.Write([]byte(info.ModTime().String()))
+		_, _ = adlerHash.Write([]byte(strconv.FormatInt(info.Size(), 16)))
+		return nil
+	})
+	if err != nil {
+		return true
+	}
+
+	newHash := adlerHash.Sum([]byte{})
+
+	if bytes.Compare(oldHash, newHash) != 0 {
+		_ = ioutil.WriteFile(filename+".hash", newHash, 0666)
```

Comment:
The hash should not be updated until after `bindata.go` is generated successfully.

Reply:
done